src/components/vr_cooperation/src/activate_service_request.cc

```diff
@@ -42,7 +42,7 @@ CREATE_LOGGERPTR_GLOBAL(logger_, "ActivateServiceRequest")
 
 ActivateServiceRequest::ActivateServiceRequest(
       VRModule* parent,
-      const application_manager::MessagePtr& message)
+      const vr_hmi_api::ServiceMessage& message)
   : BaseCommandRequest(parent, message) {
```

Comment:
ActivateServiceRequest should inheritance from RequestFromHMI, 
VRModule will responsible to use HMICommandFactory to create appropriate command, then call Run(), in this method will SendRequest() to mobile.